server/route/ClientController.js

```diff
@@ -198,4 +200,103 @@ router.delete(
 	})
 );
 
+// Route to match stylists based on client's hair details and city
+router.get("/matchStylists/:username", async (req, res) => {
+	try {
+		const { username } = req.params;
+
+		// Step 1: Retrieve client details based on username
+		const client = await Client.findOne({ username });
+
+		if (!client) {
+			return res.status(404).send({ message: "Client not found." });
+		}
+
+		// Step 2: Extract client's hair details
+		const clientHairDetails = client.hairDetails;
+
+		// Step 3: Determine city based on header or client info
+		const headerCity = req.headers["city"];
+		const cityToUse =
+			headerCity && headerCity.trim() ? headerCity : client.info.city;
+
+		if (!cityToUse) {
+			return res.status(400).send({
+				message:
+					"City is not specified in the header or client profile.",
+			});
+		}
+
+		// Step 4: Retrieve stylists from the determined city
+		const stylistsInCity = await Stylist.find({ "info.city": cityToUse });
```

Comment:
It doesn't seem like city is being populated in MongoDB because city is not part of the Stylist schema. @sohumraoGT can you also modify model/stylist.js?
![Screen Shot 2024-10-19 at 2 58 25 PM](https://github.com/user-attachments/assets/af814a1f-8214-487e-987b-9b91839048d6)

Reply:
City is a field within the info object of the account schema